OpenEphys.Onix/OpenEphys.Onix/NeuropixelsV2QuadShankProbe.cs

```diff
@@ -0,0 +1,140 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.Xml.Serialization;
+
+namespace OpenEphys.Onix
+{
+    public enum NeuropixelsV2QuadShankReference : uint
+    {
+        External,
+        Tip1,
+        Tip2,
+        Tip3,
+        Tip4
+    }
+    public enum NeuropixelsV2QuadShankBank
+    {
+        A,
+        B,
+        C,
+        D,
+    }
+
+    public class NeuropixelsV2QuadShankProbe
+    {
+        [XmlIgnore]
+        public readonly List<NeuropixelsV2QuadShankElectrode> Electrodes;
+
+        public NeuropixelsV2QuadShankProbe()
+        {
+            Electrodes = new List<NeuropixelsV2QuadShankElectrode>(NeuropixelsV2Definitions.ElectrodePerShank * 4);
+            for (int i = 0; i < NeuropixelsV2Definitions.ElectrodePerShank * 4; i++)
+            {
+                Electrodes.Add(new NeuropixelsV2QuadShankElectrode() { ElectrodeNumber = i });  
+            }
+
+            ChannelMap = new List<NeuropixelsV2QuadShankElectrode>(NeuropixelsV2Definitions.ChannelCount);
+            for (int i = 0; i < NeuropixelsV2Definitions.ChannelCount; i++)
+            {
+                ChannelMap.Add(Electrodes.Find(e => e.Channel == i));
+            }
+        }
+
+        public NeuropixelsV2QuadShankReference Reference { get; set; } = NeuropixelsV2QuadShankReference.External;
+
+        public List<NeuropixelsV2QuadShankElectrode> ChannelMap { get;  }
+
+        public void SelectElectrodes(List<NeuropixelsV2QuadShankElectrode> electrodes)
+        {
+            foreach (var e in electrodes)
+            {
+                ChannelMap[e.Channel] = e;
+            }
+        }
+    }
+
+    public class NeuropixelsV2QuadShankElectrode
+    {
+        private int electrodeNumber = 0;
+        public int ElectrodeNumber
+        {
+            get
+            {
+                return electrodeNumber;
+            }
```

Comment:
```suggestion
            get => electrodeNumber;
```

Taking advantage of new getter syntax for conciseness.